src/Coordination/SnapshotableHashTable.h

```diff
@@ -15,6 +15,19 @@ struct ListNode
     bool active_in_map;
 };
 
+template <class V>
```

Comment:
Too complex to be honest. Maybe just omit this check? Or add static assert in the constructor of SnapshotableHashTable?

Reply:
The main purpose of the check is to make the unit test compatible. If you still think it is to complex, I will remove it.